rust/chains/abacus-ethereum/src/outbox.rs

```diff
@@ -60,36 +56,42 @@ where
 {
     contract: Arc<EthereumOutboxInternal<M>>,
     provider: Arc<M>,
-    #[allow(unused)]
-    from_height: u32,
-    #[allow(unused)]
-    chunk_size: u32,
     finality_blocks: u32,
+    outbox_domain: OnceCell<u32>,
 }
 
 impl<M> EthereumOutboxIndexer<M>
 where
     M: Middleware + 'static,
 {
     /// Create new EthereumOutboxIndexer
-    pub fn new(
-        provider: Arc<M>,
-        locator: &ContractLocator,
-        from_height: u32,
-        chunk_size: u32,
-        finality_blocks: u32,
-    ) -> Self {
+    pub fn new(provider: Arc<M>, locator: &ContractLocator, finality_blocks: u32) -> Self {
+        let contract = Arc::new(EthereumOutboxInternal::new(
+            &locator.address,
+            provider.clone(),
+        ));
         Self {
-            contract: Arc::new(EthereumOutboxInternal::new(
-                &locator.address,
-                provider.clone(),
-            )),
+            contract,
             provider,
-            from_height,
-            chunk_size,
             finality_blocks,
+            outbox_domain: OnceCell::new(),
```

Comment:
instead of needing this & reading this from the contract, we can just use the `locator.domain` that's passed into `fn new`

Reply:
In my defense this was still an improvement lol. Good catch and changed.